docs/documentation/upgrading/topics/keycloak/changes-22_0_0.adoc

```diff
@@ -195,3 +190,36 @@ Up to this version, the resolving of fallback messages was inconsistent across t
 The implementation has now been unified for all themes. In general, the message for the most specific matching language tag has the highest priority. If there are both a realm localization message and a Theme 18n message, the realm localization message has the higher priority. Summarized, the priority of the messages is as follows  (RL = realm localization, T = Theme i18n files): `RL <variant> > T <variant> > RL <region> > T <region> > RL <language> > T <language> > RL en > T en`.
 
 Probably this can be better explained with an example: When the variant `de-CH-1996` is requested and there is a realm localization message for the variant, this message will be used. If such a realm localization message does not exist, the Theme i18n files are searched for a corresponding message for that variant. If such a message does not exist, a realm localization message for the region (`de-CH`) will be searched. If such a realm localization message does not exist, the Theme i18n files are searched for a message for that region. If still no message is found, a realm localization message for the language (`de`) will be searched. If there is no matching realm localization message, the Theme i18n files are be searched for a message for that language. As last fallback, the English (`en`) translation is used: First, an English realm localization will be searched - if not found, the Theme 18n files are searched for an English message.
+
+== Deprecation of Keycloak OpenID Connect Adapters
+
+From this release, we are no longer investing our time on some of Keycloak OpenID Connect Adapters:
+
+* Keycloak Wildfly OpenID Connect Adapter
+* Keycloak JEE Servlet OpenID Connect Adapter
+* Keycloak Spring Boot and Spring Security OpenID Connect Adapter
+
+This move is already reflected in our documentation and in our quickstart repository. Please, consider look at the following
+references for more information:
+
+* link:https://github.com/keycloak/keycloak-quickstarts[Keycloak Quickstart GitHub Repository]
+* link:https://www.keycloak.org/docs/latest/securing_apps/[Keycloak Securing Applications Documentation]
+
+We recommend to start looking at migrating your applications to the alternatives available from the references above as those
```

Comment:
Not an issue, only a suggestion:

"We recommend starting to look into moving your applications to the alternatives from the references above. Those adapters should not be available anymore in future releases."